### PR TITLE
Add aws-c-common/0.4.25 recipe

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **aws-c-common/0.4.25**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/aws-c-common/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
+from conans import CMake, ConanFile, tools
+import os
+
+
+class AwsCCommon(ConanFile):
+    name = "aws-c-common"
+    description = "Core c99 package for AWS SDK for C. Includes cross-platform primitives, configuration, data structures, and error handling."
+    topics = ("conan", "aws", "amazon", "cloud", )
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/aws/aws-sdk-cpp"
```
Comment (uilianries) [inline recipes/aws-c-common/all/conanfile.py]:
```suggestion
    homepage = "https://github.com/awslabs/aws-c-common"
```

File: recipes/aws-c-common/all/conandata.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+sources:
+  "0.4.25":
```
Comment (uilianries) [inline recipes/aws-c-common/all/conandata.yml]: Is there any reason for the version 0.4.25? The latest version available is 0.4.34

Comment (madebr) [reply recipes/aws-c-common/all/conandata.yml]: Those are pre-releases, so I just used the last *real* release.

**Comment by madebr** [comment]
> The author says:
> 
> > Always enable position independent code, since this code will always end up in a shared lib
> > list(APPEND AWS_C_FLAGS -fPIC)
> 
> https://github.com/awslabs/aws-c-common/blob/v0.4.34/cmake/AwsCFlags.cmake#L66
> 
> So I think we can't remove that fPIC, as Conan does it automatically and we offer static lib version.

I think we can remove the line.
If consumers disable fPIC and use it in a shared library, then their linker will complain loudly.
The comment in their README is valid for all software. If fPIC is disabled, then it can't be used in shared libraries. So I don't see why we would treat this package different.

**Pull request merged**